### DISCUSSION TITLE
The problem of not executing the code (error 401)

### DISCUSSION THREAD

**Discussion by seyedrezabazyar** [tectalichq/public-openai-client-php#37] [General]
I ran the following code

```
<?php

require 'vendor/autoload.php';

$openaiClient = \Tectalic\OpenAi\Manager::build(
    new \GuzzleHttp\Client(),
    new \Tectalic\OpenAi\Authentication(getenv('sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'))
);

/** @var \Tectalic\OpenAi\Models\ChatCompletions\CreateResponse $response */
$response = $openaiClient->chatCompletions()->create(
    new \Tectalic\OpenAi\Models\ChatCompletions\CreateRequest([
        'model' => 'gpt-4',
        'messages' => [
            [
                'role' => 'user',
                'content' => 'Will using a well designed and supported third party package save time?'
            ],
        ],
    ])
)->toModel();

echo $response->choices[0]->message->content;
```

But I got this error while running:
`Fatal error: Uncaught Tectalic\OpenAi\ClientException: Unsuccessful response. HTTP status code: 401 (Unauthorized). in /private/var/www/chatgpt/vendor/tectalic/openai/src/Handlers/ChatCompletions.php:150 Stack trace: #0 /private/var/www/chatgpt/index.php(21): Tectalic\OpenAi\Handlers\ChatCompletions->toModel() #1 {main} thrown in /private/var/www/chatgpt/vendor/tectalic/openai/src/Handlers/ChatCompletions.php on line 150`

My api is completely correct and I changed it several times but still it doesn't work.
I am using the free version of chatgpt and I want to test it and if it works, I will buy the premium account of chatgpt.

How can I fix this problem?

**Comment by the-csaba** [comment]
Hi @seyedrezabazyar 

It appears that your OpenAI key may not have been loaded correctly. 

Could you add the line `echo getenv('sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');` and inspect the output? This should return the actual key.

For best practices, we recommend storing your `OPENAI_API_KEY` as an environment variable instead of hard-coding it directly into your project. This approach enhances security and maintainability. However, you're welcome to adopt a different method if it suits your needs better.

**Comment by seyedrezabazyar** [comment]
I removed getenv from the code and the problem was solved
```new \Tectalic\OpenAi\Authentication('sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx))```

But the texts produced through API have a much lower quality than the ChatGPT website :(
